spec/person_spec.rb

```diff
@@ -0,0 +1,27 @@
+require './people/person'
+
+describe Person do
+  context 'when initializing with valid parameters' do
+    before :each do
+      @person = Person.new(1, 'name', 17, true)
+    end
+
+    it 'returns a valid Person object when given four arguments' do
+      expect(@person).to be_instance_of(Person)
+    end
+
+    it "correct_name method returns the person's name" do
+      expect(@person.correct_name).to eq(@person.name)
+    end
+
+    it 'can_use_services? method returns true if the person is of age or has parent_permission' do
+      expect(@person.can_use_services?).to eq(true)
+    end
+  end
+
+  context 'when initializing with invalid parameters' do
+    it 'raises an ArgumentError when not given three or four arguments' do
+      expect { Person.new(1, 'name') }.to raise_exception(ArgumentError)
+    end
+  end
+end
```

Comment:
- [ ] Good job so far, but you just need to add some tests and you're good to go 👍🏾 
  - [ ] You should add a test in the context when creating a new instance of the Person class:
    - A scenario of the constructor method returns a Person object with provided age, name, and empty rentals,
    - A scenario of the constructor method sets the default value for the optional name parameter if it is not provided
  - [ ] You should add a test in the context when adding a new rentals:
    - A scenario of the add_rental method returns a rental and adds it to the book
  - [ ] You should add a test in the context when checking permissions:
    - A scenario of the `can_use_services?` method returns false if under age and no parent permission,
    - A scenario of the `can_use_services?` method returns true if over age,
  - [ ] You should add a test in the context when validating name:
    - A scenario of the correct_name method corrects the name, for example, if the name is `alongnameiscorrected` it should be equal to `Alongnamei`

Reply:
Hello @cynthiainga I've added the test you've asked except for this test, *A scenario of the constructor method sets the default value for the optional name parameter if it is not provided* as I've long changed the name parameter to be required. Hopefully you can understand the situation, I've even made sure to quickly and effectively add the other test to demonstrate our understanding of the topic, thank you for reading and have a nice day. 🙂